patroni/dcs/__init__.py

```diff
@@ -912,13 +926,22 @@ def __permanent_slots(self) -> Dict[str, Union[Dict[str, Any], Any]]:
                     ret[name] = {}
                 if isinstance(ret[name], dict):
                     ret[name]['lsn'] = lsn
+
+        # there is no slot on the leader for itself, use `lsn` from the member key.
+        leader = self.leader and self.leader.member
+        if leader and leader.lsn:
+            name = slot_name_from_member_name(leader.name)
+            if name in ret and self.is_physical_slot(ret[name]):
+                if not isinstance(ret[name], dict):
+                    ret[name] = {}
+                if 'lsn' not in ret[name]:
+                    ret[name]['lsn'] = leader.lsn
```

Comment:
Why do we update only if `'lsn' not in ret[name]'`?
In the previous part, which handles the non-leader slot, we always update the LSN.
The above code could face issues if by chance ones does something like this in the config:

```
slots:
  postgresql0:
    lsn: 123
    type: physical
  postgresql1:
    type: physical
  postgresql2:
    type: physical
```

Unlikely to occur, but an open door 😆